src/region_growing.jl

```diff
@@ -0,0 +1,208 @@
+
+default_diff_fn{CT1<:Colorant, CT2<:Colorant}(c1::CT1,c2::CT2) = sqrt(sum(abs2,(c1)-Images.accum(CT2)(c2)))
+
+"""
+    seg_img = seeded_region_growing(img, seeds, [kernel_dim], [diff_fn])
+    seg_img = seeded_region_growing(img, seeds, [neighbourhood], [diff_fn])
+
+Segments the N-D image `img` using the seeded region growing algorithm
+and returns a [`SegmentedImage`](@ref) containing information about the segments.
+
+# Arguments:
+* `img`             :  N-D image to be segmented (arbitrary indices are allowed)
+* `seeds`           :  `Vector` containing seeds. Each seed is a Tuple of a
+                       CartesianIndex{N} and a label. See below note for more
+                       information on labels.
+* `kernel_dim`      :  (Optional) `Vector{Int}` having length N whose ith element
+                       is an odd positive integer representing the length of the
+                       ith edge of the N-orthotopic neighbourhood
+* `neighbourhood`   :  (Optional) Function taking CartesianIndex{N} as input and
+                       returning the neighbourhood of that point.
+* `diff_fn`         :  (Optional) Function that returns a difference measure(δ)
+                       between a region mean and a point
```

Comment:
Perhaps clarify that it operates on the values, e.g., "...between the color of a point and the mean color of a region" or something.